frontend/ts-src/lib/network.ts

```diff
@@ -0,0 +1,124 @@
+/* Corna frontend network utilities. */
+
+// axios types: https://github.com/axios/axios/blob/v1.x/index.d.ts
+import axios from "axios";
+import {
+    AxiosError,
+    AxiosHeaders,
+    AxiosPromise,
+    AxiosRequestConfig,
+    AxiosResponse,
+} from "axios";
+
+import { handlePromise } from "./utils.js";
+
+
+export function getApiUrl(): string {
+    /* Get the correct API URL.
```

Comment:
Does this work? Usually JSDoc is above the function declaration. If you put it above the variable it will bind to that for hinting

Reply:
I'm not sure tbh, I dont use an IDE so I don't get any typehints/errors till I run the compiler.

Reply:
You what? Are you using notepad? You won't get errors because it's a comment and optional spec. It's also valid, but will probably show up in the wrong place